packages/service-provider-sfa/src/SfaServiceProvider.ts

```diff
@@ -47,10 +55,10 @@ class SfaServiceProvider extends ServiceProviderBase {
 
   async connect(params: LoginParams): Promise<BN> {
     const { verifier, verifierId, idToken, subVerifierInfoArray } = params;
-    const verifierDetails = { verifier, verifierId };
+    this.verifierDetails = { verifier, verifierId };
 
     // fetch node details.
-    const { torusNodeEndpoints, torusNodePub, torusIndexes } = await this.nodeDetailManagerInstance.getNodeDetails(verifierDetails);
+    const { torusNodeEndpoints, torusNodePub, torusIndexes } = await this.nodeDetailManagerInstance.getNodeDetails(this.verifierDetails);
```

Comment:
We can just save the torusNodeEndpoints 's origin as private and use it during the as the `HostUrl` (or storageHostUrl )